### PR TITLE
[INLONG-6061][Sort] Support TDSQL-Kafka connector

### LINKED ISSUE #6061: [Feature][Sort] Support CDC  for Tencent Cloud's TDSQL distribution database
### Description

Tencent Cloud's TDSQL distribution database can be captured changed data by [DTS](https://cloud.tencent.com/document/product/571/68060) . TDSQL's binlog will be sent into DTS Kafka, then we can subsribe change data from Kafka.

This feature lets inlong sort to generate TDSQL Kafka Flink SQL semantics.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liaorui**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #6061

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/format/ProtobufFormat.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public ProtobufFormat(@JsonProperty(value = "codec") String codec) {
+        this.codec = codec;
+    }
+
+    @JsonCreator
+    public ProtobufFormat() {
+    }
+
+    /**
+     * Return avro
```
Comment (EMsnap) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/format/ProtobufFormat.java:51]: doc doesn't match code

File: inlong-common/src/main/java/org/apache/inlong/common/enums/TaskTypeEnum.java [hunk 2]
```diff
@@ -31,8 +31,10 @@ public enum TaskTypeEnum {
     MONGODB(9),
     TUBEMQ(10),
     REDIS(11),
-
-
+    CKAFKA(12),
+    MYSQL(13),
+    TDSQL_KAFKA(14),
+    DORIS(15)
```
Comment (healchow) [inline inlong-common/src/main/java/org/apache/inlong/common/enums/TaskTypeEnum.java:37]:
```suggestion
    DORIS(15),


```

File: inlong-common/src/main/java/org/apache/inlong/common/enums/TaskTypeEnum.java [hunk 3]
```diff
@@ -31,8 +31,10 @@ public enum TaskTypeEnum {
     MONGODB(9),
     TUBEMQ(10),
     REDIS(11),
-
-
+    CKAFKA(12),
+    MYSQL(13),
```
Comment (healchow) [inline inlong-common/src/main/java/org/apache/inlong/common/enums/TaskTypeEnum.java:35]: Excuse me, what does the difference between the `SQL`, `BINLOG`, and `MYSQL`?

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/format/ProtobufFormat.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonIgnore;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonTypeName;
+
+/**
+ * The protobuf format.
+ *
+ */
+@JsonTypeName("protobufFormat")
+@Data
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/format/ProtobufFormat.java:33]:
```suggestion
 */
@Data
@JsonTypeName("protobufFormat")
```

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/TdsqlKafkaConstant.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/*
+ *   Licensed to the Apache Software Foundation (ASF) under one
+ *   or more contributor license agreements.  See the NOTICE file
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/TdsqlKafkaConstant.java:3]: Please keep this license header the same as other files.

File: inlong-common/src/main/java/org/apache/inlong/common/enums/TaskTypeEnum.java (hunk 3 above)
Comment (liaorui) [reply inlong-common/src/main/java/org/apache/inlong/common/enums/TaskTypeEnum.java:35]: delete MYSQL in line 35

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TdsqlKafkaExtractNode.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.inlong.sort.protocol.node.format.Format;
+import org.apache.inlong.sort.protocol.node.format.ProtobufFormat;
+import org.apache.inlong.sort.protocol.transformation.WatermarkField;
+
+/**
+ * Kafka extract node for extract data from kafka
+ */
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeName("tdsqlKafkaExtract")
+public class TdsqlKafkaExtractNode extends KafkaExtractNode implements InlongMetric, Metadata, Serializable {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TdsqlKafkaExtractNode.java:44]: Tdsql -> TDSQL

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/TdsqlKafkaConstant.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+package org.apache.inlong.sort.protocol.constant;
+
+/**
+ * tdsql kafka option constant
+ *
+ * @see <a href="https://cloud.tencent.com/document/product/849/71448"></a>
+ */
+public class TdsqlKafkaConstant {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/TdsqlKafkaConstant.java:26]: Tdsql -> TDSQL

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TdsqlKafkaExtractNode.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        if (getFormat() instanceof ProtobufFormat) {
+            options.put(TdsqlKafkaConstant.CONNECTOR, TdsqlKafkaConstant.TDSQL_SUBSCRIBE);
+            options.put(TdsqlKafkaConstant.SCAN_STARTUP_MODE, getKafkaScanStartupMode().getValue());
+            if (StringUtils.isNotEmpty(getScanSpecificOffsets())) {
+                options.put(TdsqlKafkaConstant.SCAN_STARTUP_SPECIFIC_OFFSETS, getScanSpecificOffsets());
+            }
+            options.putAll(getFormat().generateOptions(false));
+        } else {
+            throw new IllegalArgumentException("kafka extract node format is IllegalArgument");
+        }
```
Comment (gong) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TdsqlKafkaExtractNode.java:86]: description should be `tdsql kafka extract node`

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TdsqlKafkaExtractNode.java [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+        if (StringUtils.isNotEmpty(getGroupId())) {
+            options.put(TdsqlKafkaConstant.PROPERTIES_GROUP_ID, getGroupId());
+        }
+        for (String key : getProperties().keySet()) {
+            options.put(key, getProperties().get(key));
+        }
```
Comment (gong) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TdsqlKafkaExtractNode.java:92]: It is duplication with code lines 72-74
```java
  if (getProperties() != null && !getProperties().isEmpty()) {
            options.putAll(getProperties());
        }
```

**Review by gong — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TDSQLKafkaExtractNode.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+import org.apache.inlong.sort.protocol.InlongMetric;
+import org.apache.inlong.sort.protocol.Metadata;
+import org.apache.inlong.sort.protocol.constant.TDSQLKafkaConstant;
+import org.apache.inlong.sort.protocol.enums.KafkaScanStartupMode;
+import org.apache.inlong.sort.protocol.node.format.Format;
+import org.apache.inlong.sort.protocol.node.format.ProtobufFormat;
+import org.apache.inlong.sort.protocol.transformation.WatermarkField;
+
+/**
+ * Kafka extract node for extract data from kafka
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TDSQLKafkaExtractNode.java:40]:
```suggestion
 * TDSQK Kafka extract node info
```

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TDSQLKafkaExtractNode.java [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+        if (getFormat() instanceof ProtobufFormat) {
+            options.put(TDSQLKafkaConstant.CONNECTOR, TDSQLKafkaConstant.TDSQL_SUBSCRIBE);
+            options.put(TDSQLKafkaConstant.SCAN_STARTUP_MODE, getKafkaScanStartupMode().getValue());
+            if (StringUtils.isNotEmpty(getScanSpecificOffsets())) {
+                options.put(TDSQLKafkaConstant.SCAN_STARTUP_SPECIFIC_OFFSETS, getScanSpecificOffsets());
+            }
+            options.putAll(getFormat().generateOptions(false));
+        } else {
+            throw new IllegalArgumentException("tdsql kafka extract node format is IllegalArgument");
+        }
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TDSQLKafkaExtractNode.java:86]:
```suggestion
        if (!(getFormat() instanceof ProtobufFormat)) {
            throw new IllegalArgumentException("tdsql kafka extract node format is IllegalArgument");
        }

        options.put(TDSQLKafkaConstant.CONNECTOR, TDSQLKafkaConstant.TDSQL_SUBSCRIBE);
        options.put(TDSQLKafkaConstant.SCAN_STARTUP_MODE, getKafkaScanStartupMode().getValue());
        if (StringUtils.isNotEmpty(getScanSpecificOffsets())) {
            options.put(TDSQLKafkaConstant.SCAN_STARTUP_SPECIFIC_OFFSETS, getScanSpecificOffsets());
        }
        options.putAll(getFormat().generateOptions(false));
```

File: inlong-common/src/main/java/org/apache/inlong/common/enums/TaskTypeEnum.java [hunk 12]
```diff
@@ -31,6 +31,9 @@ public enum TaskTypeEnum {
     MONGODB(9),
     TUBEMQ(10),
     REDIS(11),
+    CKAFKA(12),
+    TDSQL_KAFKA(14),
```
Comment (healchow) [inline inlong-common/src/main/java/org/apache/inlong/common/enums/TaskTypeEnum.java:35]: Why no `13` here?

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/TDSQLKafkaConstant.java [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sort.protocol.constant;
+
+/**
+ * tdsql kafka option constant
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/TDSQLKafkaConstant.java:21]:
```suggestion
 * TDSQL Kafka option constant
```

**Pull request closed without merging**